python/ctsm/toolchain/gen_mksurfdata_namelist.py

```diff
@@ -388,34 +566,247 @@ def main():
         sys.exit(error_msg)
 
     logger.info("pft_years = %s", pft_years)
-
-    # Create land-use txt file for a transient case.
-    # Determine the run type and if a transient run create output landuse txt file
-    if end_year > start_year:
-        run_type = "transient"
+    return pft_years_ssp, pft_years
+
+
+def write_nml_outdata(
+    nosurfdata_flag,
+    vic_flag,
+    inlandwet,
+    glc_flag,
+    hostname,
+    logname,
+    num_pft,
+    fdyndat,
+    fsurdat,
+    fsurlog,
+    gitdescribe,
+    nlfile,
+):
+    """
+    Write output namelist file: output data
+    """
+    # -------------------
+    # output data files
+    # -------------------
+    if nosurfdata_flag:
+        nlfile.write("  fsurdat = ' ' \n")
     else:
-        run_type = "timeslice"
-    logger.info("run_type  = %s", run_type)
+        nlfile.write(f"  fsurdat = '{fsurdat}'\n")
+    nlfile.write(f"  fsurlog = '{fsurlog}' \n")
+    nlfile.write(f"  fdyndat = '{fdyndat}' \n")
+
+    # -------------------
+    # output data logicals
+    # -------------------
+    nlfile.write(f"  numpft = {num_pft} \n")
+    nlfile.write(f"  no_inlandwet = .{str(not inlandwet).lower()}. \n")
+    nlfile.write(f"  outnc_3dglc = .{str(glc_flag).lower()}. \n")
+    nlfile.write(f"  outnc_vic = .{str(vic_flag).lower()}. \n")
+    nlfile.write("  outnc_large_files = .false. \n")
+    nlfile.write("  outnc_double = .true. \n")
+    nlfile.write(f"  logname = '{logname}' \n")
+    nlfile.write(f"  hostname = '{hostname}' \n")
+    nlfile.write(f"  gitdescribe = '{gitdescribe}' \n")
+
+
+def write_nml_rawinput(
+    start_year,
+    force_model_mesh_file,
+    force_model_mesh_nx,
+    force_model_mesh_ny,
+    vic_flag,
+    rawdata_files,
+    landuse_fname,
+    mksrf_ftopostats_override,
+    nlfile,
+    must_run_download_input_data,
+):
+    """
+    Write output namelist file: raw input data
+    """
+    # pylint: disable=too-many-statements
```

Comment:
This one must be really close to the limit. Perhaps we should increase it?

Reply:
I think I like the limit being kept at 50, and then we can have a little bit of wiggle room. If we increase it, we'll always be tempting to add wiggle room on top of that.

Reply:
Yes, the sliding down the hill problem, it keeps sliding toward infinity. I was thinking the same. I suppose 50 is typically a lot for python. It's a guideline for having code that's too complex and can't be unit tested. So yes the longer we make it the longer we'll want to keep making it. I suppose that default of 50 has something behind it as well.

Reply:
OK, resolving.